src/Components/Popover/Popover.tsx

```diff
@@ -0,0 +1,476 @@
+// Libraries
+import React, {Component, ReactChild, createRef, CSSProperties} from 'react'
+import classnames from 'classnames'
+
+// Components
+import {ClickOutside} from '../ClickOutside/ClickOutside'
+import {PopoverDismissButton} from './PopoverDismissButton'
+
+// Styles
+import './Popover.scss'
+
+// Types
+import {
+  ComponentColor,
+  StandardProps,
+  PopoverInteraction,
+  PopoverPosition,
+  PopoverType,
+} from '../../Types'
+
+interface Props extends StandardProps {
+  /** Popover dialog color */
+  color: ComponentColor
+  /** Popover dialog contents */
+  contents: (onHide?: () => void) => ReactChild
+  /** Type of interaction to show the popover dialog */
+  showEvent: PopoverInteraction
+  /** Type of interaction to hide the popover dialog */
+  hideEvent: PopoverInteraction
+  /** Pixel distance between trigger and popover dialog */
+  distanceFromTrigger: number
+  /** Where to position the popover relative to the trigger (assuming it fits there) */
+  position: PopoverPosition
+  /** Means of applying color to popover */
+  type: PopoverType
+  /** Renders the popover dialog visible initially */
+  initiallyVisible: boolean
+}
+
+interface State {
+  expanded: boolean
+  triggerRect: ClientRect | DOMRect | null
+}
+
+interface PopoverFlush {
```

Comment:
Although this is internal to the Popover component, I'm not sure if it's the most understandable name

Reply:
A comment with an example might help.